spec.md

```diff
@@ -523,13 +518,295 @@ schema being used.
         "bullets": [
           "40 concurrent connections"
         ]
+      },
+      "schemas": {
+        "service_instance": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+            }
+          },
+          "update": {
+            "parameters": {
+              "$schema": "http://json-schema.org/draft-04/schema#",
+              "type": "object",
+              "propeties": {
+                "some-other-account": {
+                  "description": "Other fake account needed for update.",
+                  "type": "string"
+                },
+                "billing": {
+                  "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+                }
+              }
+            }
+          }
+        },
+        "service_binding": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1/bindings#AccountParameters"
+            }
+          }
+        }
       }
     }]
   }]
 }
 ```
 
 
+## JSON Schemas
+
+The JSON Schemas endpoint is used to define reusable definitions of JSON Schema
+Objects referenced throughout the [Catalog](#catalog-management). It will be the
+responsibility of the Platform to examine the response from the catalog to
```

Comment:
I believe the Service Broker will be control over whether this feature is used or not. Meaning, Platforms must look for $ref, support osbapi:// URLs AND support retrieving things from a separate JSON Schema Endpoint if their Service Brokers choose to use these features and the Platform does schema checking. This basically makes it a requirement on Platforms that want to deal with schema at all.

I think this means that we should probably add some MUSTs about what the Platform needs to support *if* it wants to do schema validation. Although, I guess it could reject the request if it doesn't support these things - but then it must be at least aware enough to even look for things like osbapi:// to reject the request. Anyway, I think we need to say something about this to ensure some kind of interop and validation during catalog processing, because I think it will cause issues if the Broker returns something that the Platform doesn't know about (or support) and no one detects this mismatch.